test/print_prefix.lean

```diff
@@ -48,10 +74,72 @@ TestStruct.noConfusion : {P : Sort u} → {v1 v2 : TestStruct} → v1 = v2 → T
 TestStruct.noConfusionType : Sort u → TestStruct → TestStruct → Sort u
 TestStruct.rec : {motive : TestStruct → Sort u} → ((foo bar : Int) → motive { foo := foo, bar := bar }) → (t : TestStruct) → motive t
 TestStruct.recOn : {motive : TestStruct → Sort u} → (t : TestStruct) → ((foo bar : Int) → motive { foo := foo, bar := bar }) → motive t
+-/
+#guard_msgs in
+#print prefix (config:={propositions:=false}) TestStruct
+
+/--
+info: TestStruct.mk.inj : ∀ {foo bar foo_1 bar_1 : Int}, { foo := foo, bar := bar } = { foo := foo_1, bar := bar_1 } → foo = foo_1 ∧ bar = bar_1
+TestStruct.mk.injEq : ∀ (foo bar foo_1 bar_1 : Int),
+  ({ foo := foo, bar := bar } = { foo := foo_1, bar := bar_1 }) = (foo = foo_1 ∧ bar = bar_1)
+TestStruct.mk.sizeOf_spec : ∀ (foo bar : Int), sizeOf { foo := foo, bar := bar } = 1 + sizeOf foo + sizeOf bar
+-/
+#guard_msgs in
+#print prefix (config:={propositionsOnly:=true}) TestStruct
+
+/--
+info: TestStruct
+TestStruct.bar
+TestStruct.casesOn
+TestStruct.foo
+TestStruct.mk
+TestStruct.mk.inj
+TestStruct.mk.injEq
+TestStruct.mk.sizeOf_spec
+TestStruct.noConfusion
+TestStruct.noConfusionType
+TestStruct.rec
+TestStruct.recOn
+-/
+#guard_msgs in
+#print prefix (config:={showTypes:=false}) TestStruct
+
+/--
+info: TestStruct : Type
+TestStruct._sizeOf_1 : TestStruct → Nat
+TestStruct._sizeOf_inst : SizeOf TestStruct
+TestStruct.bar : TestStruct → Int
+TestStruct.casesOn : {motive : TestStruct → Sort u} → (t : TestStruct) → ((foo bar : Int) → motive { foo := foo, bar := bar }) → motive t
+TestStruct.foo : TestStruct → Int
+TestStruct.mk : Int → Int → TestStruct
 TestStruct.mk.inj : ∀ {foo bar foo_1 bar_1 : Int}, { foo := foo, bar := bar } = { foo := foo_1, bar := bar_1 } → foo = foo_1 ∧ bar = bar_1
 TestStruct.mk.injEq : ∀ (foo bar foo_1 bar_1 : Int),
   ({ foo := foo, bar := bar } = { foo := foo_1, bar := bar_1 }) = (foo = foo_1 ∧ bar = bar_1)
 TestStruct.mk.sizeOf_spec : ∀ (foo bar : Int), sizeOf { foo := foo, bar := bar } = 1 + sizeOf foo + sizeOf bar
+TestStruct.noConfusion : {P : Sort u} → {v1 v2 : TestStruct} → v1 = v2 → TestStruct.noConfusionType P v1 v2
+TestStruct.noConfusionType : Sort u → TestStruct → TestStruct → Sort u
+TestStruct.rec : {motive : TestStruct → Sort u} → ((foo bar : Int) → motive { foo := foo, bar := bar }) → (t : TestStruct) → motive t
+TestStruct.recOn : {motive : TestStruct → Sort u} → (t : TestStruct) → ((foo bar : Int) → motive { foo := foo, bar := bar }) → motive t
 -/
 #guard_msgs in
-#print prefix TestStruct
+#print prefix (config:={internals:=true}) TestStruct
```

Comment:
Absolute final nitpick! It would be nice if this example also tested `proof_` and `match_`.

Reply:
I've added a more comprehensive test case for printing internals, but I am not sure the positive test of checking the output of `#print prefix (config:={internals:=true})` is a good idea.  The problem is that internal names are inherently subject to change.

I've done it anyways with a warning this may be volatile.  If maintaining the test case is a chore, then I think it should be removed.